nucypher/blockchain/eth/clients.py

```diff
@@ -217,6 +222,15 @@ def send_transaction(self, transaction: dict) -> str:
     def send_raw_transaction(self, transaction: bytes) -> str:
         return self.w3.eth.sendRawTransaction(raw_transaction=transaction)
 
+    def _has_latest_block(self):
+        # check that our local chain data is up to date
+        return (
+            datetime.datetime.now() -
+            datetime.datetime.fromtimestamp(
+                self.w3.eth.getBlock(self.w3.eth.blockNumber)['timestamp']
+            )
+        ).seconds < 30
```

Comment:
Looks like the comment got lost, so repeating here. This one doesn't work properly because this condition is satisfied also for "1 day and 3 seconds" or "10 days and 5 seconds". Replace with:
```
            (datetime.datetime.now() -
            datetime.datetime.fromtimestamp(
                self.w3.eth.getBlock(self.w3.eth.blockNumber)['timestamp'])) < timedelta(seconds=30)
```
or
```
time.time() - self.w3.eth.getBlock(self.w3.eth.blockNumber)['timestamp'] < 30
```

Reply:
I think using `.total_seconds()` instead of `.seconds` may solve the problem. See my comment below in the unit test file and https://docs.python.org/3/library/datetime.html